### ISSUE TITLE
No contrast information available

### ISSUE THREAD

**Issue by Th3S4mur41** [MicrosoftEdge/DevTools#54]
In many cases, the text color info doesn't show contrast information.
Tested in Edge 102.0.1249.0 (Official build) canary (64-bit) with both the regular WCAG AA/AAA guidelines and the experimental APCA ratio.

![image](https://user-images.githubusercontent.com/5543595/167615552-843a739f-4374-4146-8226-2d4942bac7d4.png)

Chrome 103.0.5054.0 (Official Build) canary (64-bit) has the same issue.
Firefox 101.0b4 (64 bits) on the other hand is showing a contrast ratio as expected for the same content
![image](https://user-images.githubusercontent.com/5543595/167615801-7b542cf1-2c8f-4ef7-bac6-b2ce3bf4827f.png)

**Comment by captainbrosset** [comment]
Here is one test case where contrast calculation fails, but there might be others:

1. Open the following URL `data:text/html,<style>div::after {content: " World!";color: lightcyan;}</style><div>Hello`
2. Select the `::after` pseudo-element in the Elements panel
3. Click on the color swatch next to the `color` property in the Styles panel

<img width="331" alt="image" src="https://user-images.githubusercontent.com/1152698/167803698-fd3c8491-e36f-4c2c-9209-f336a5bb4cbd.png">

**Comment by Th3S4mur41** [comment]
The problem doesn't seem to be limited to pseudo element...
I could reproduce it on custom checkbox using a `<span>` within a `<label>`

![image](https://user-images.githubusercontent.com/5543595/168015245-6f6f45e2-8a62-4da3-b71e-82988eef4c81.png)

I also tried to set the background-color on the `<span>` directly, without impact on the problem

**Comment by captainbrosset** [comment]
Yeah, the current logic checks if the element only has 1 child node and if it is a text node. Because the `<span>` contains 2 child nodes (comment node and text node), then the logic fails.
I'll add your test case to the chromium bug. Thanks!

**Comment by captainbrosset** [comment]
The upstream bug has been merged into this other one: https://bugs.chromium.org/p/chromium/issues/detail?id=1105775
No recent activity on it unfortunately.
